Sources/SWBTaskExecution/TaskActions/BuildDependencyInfoTaskAction.swift

```diff
@@ -0,0 +1,58 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+public import SWBCore
+internal import SWBMacro
+internal import SWBProtocol
+import SWBUtil
+
+public final class BuildDependencyInfoTaskAction: TaskAction {
+    public override class var toolIdentifier: String {
+        return "build-dependency-info"
+    }
+
+    public override func performTaskAction(_ task: any ExecutableTask, dynamicExecutionDelegate: any DynamicTaskExecutionDelegate, executionDelegate: any TaskExecutionDelegate, clientDelegate: any TaskExecutionClientDelegate, outputDelegate: any TaskOutputDelegate) async -> CommandResult {
+        let commandLine = Array(task.commandLineAsStrings)
+        guard commandLine.count >= 1, commandLine[0] == "builtin-build-dependency-info" else {
```

Comment:
For all new task actions, can you please use argument parser?

(see ObjectLibraryAssemblerTaskAction for example)

this also makes it easier to delineate arguments using flags rather than relying too much on positionals.